### PR TITLE
Add _GROUP filter to isos post

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/33892

AC1 is tested and works. Job templates *not* matching the given group ID are ignored when creating new jobs via `POST isos`.

Not sure about AC2. Does "Builds that were triggered in the normal way" mean "jobs that were spawned before, but without explicit group filtering"? If yes, that wouldn't be completely trivial to implement because we would need to save whether the group filtering was used when spawning a job.

Also not sure about AC3. For me it sounds like I don't have to do anything further here. This PR just introduces a filter. The usual behavior for obsoletion isn't changed - just limited to the specified group.

### CODE REVIEW

**Comment by coolo** [comment]
I read AC2 as is: 

post isos ISO=myiso.iso ... -> 130 jobs for 5 job groups
post isos ISO=myiso.iso _GROUP=TheCoolOne ... -> 10 jobs for 1 job group

-> 140 scheduled jobs, none obsolete

**Comment by Martchus** [comment]
Ok, I suppose that simple filter won't do it then because the 2nd post would obsolete jobs created by the 1st post which belong to TheCoolOne.

Of course one could just use `_NO_OBSOLETE` to prevent this obsoletion if not wanted.

**Comment by coolo** [comment]
perhaps that should be the default for _GROUP then (or in general if you ask certain people)

**Comment by Martchus** [comment]
@foursixnine  Thinking about that I'd like to reconsider AC2. Making `_NO_OBSOLETE` default would be inconsistent. At least to me it is not obvious that adding a filter will change the obsoletion behavior. And it would require introducing yet another parameter to enable obsoletion if required after all.

@coolo From your example I read that you expect the _GROUP parameter to take a group name rather than a group ID. I could change that if you like (but @okurz knows all groups by ID anyways).

**Comment by AdamWill** [comment]
Thanks for the CC. This would be a very useful feature indeed! I kinda agree with @Martchus that it doesn't really make sense to flip the default for `_NO_OBSOLETE` just because this new behaviour is used; sure, it's *envisaged* as being for 'testing' purposes, but there's no law saying it will only be used for that. I'm not entirely sure what @coolo means by "let's require it for now (as in add it to the docu where you mention _GROUP) - and make it default" - can you elaborate on that?

Also removing interactive mode == sadface :/ But that's out of scope here I guess.

**Comment by coolo** [comment]
I mean: for this PR accept that you need to pass _NO_OBSOLETE too if you want to use the feature as specified. And in a 2nd step make NO_OBSOLETE default for all - and introduce a new variable obviously.

For interactive mode, check https://progress.opensuse.org/issues/14818#note-18

**Comment by Martchus** [comment]
It now allows to use the group name via `_GROUP` and the ID via `_GROUP_ID`. `_NO_OBSOLETE` is still required for AC2 as discussed.

File: docs/UsersGuide.asciidoc [hunk 1]
```diff
@@ -224,6 +224,11 @@ mean existing jobs for earlier builds with the same DISTRI and VERSION are
 no longer interesting, but you still want to be able to re-submit jobs for a
 build and have existing jobs for the exact same build obsoleted.
 
+_GROUP:: Job templates *not* matching the given group name are ignored. Does *not*
```
Comment (okurz) [inline docs/UsersGuide.asciidoc]: Should be easy to make this a comma split list of regexes, WDYT?

Comment (coolo) [reply docs/UsersGuide.asciidoc]: Sounds like a different user story.

**Pull request merged**